### ISSUE TITLE
"could not initialize a memory descriptor" error using tensorflow on windows using CMake

### ISSUE THREAD

**Issue by syagev** [tensorflow/tensorflow#21334]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: NO
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10 (build 17134) 
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.9
- **Python version**: 3.6
- **Bazel version (if compiling from source)**: 0.15
- **GCC/Compiler version (if compiling from source)**: CMake > VS 2017
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: `Session::run()`

### Describe the problem

I've built tensorflow (r1.9, using the CMake tools) linked with MKL (v2018 U3) and mkl-dnn (v0.15). I'm running windows 10 (build 17134) on an Intel Core i7-7820HQ CPU. 

I've built mkl-dnn from source and it's test are passing. However, a C++ project that loads a pre-trained tensorflow graph and passes an image for inference gives the following error when calling `Session::run(...)` :
```
W d:\dev\tensorflow\tensorflow\core\framework\op_kernel.cc:1318] OP_REQUIRES failed at mkl_conv_ops.cc:888 : Aborted: Operation received an exception:Status: 3, message: could not initialize a memory descriptor, in file d:\dev\tensorflow\tensorflow\core\kernels\mkl_conv_ops.cc:886
Aborted: Operation received an exception:Status: 3, message: could not initialize a memory descriptor, in file d:\dev\tensorflow\tensorflow\core\kernels\mkl_conv_ops.cc:886
         [[Node: conv1/BiasAdd = _MklConv2DWithBias[T=DT_FLOAT, _kernel="MklOp", data_format="NHWC", dilations=[1, 1, 1, 1], padding="VALID", strides=[1, 2, 2, 1], use_cudnn_on_gpu=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](conv1_pad/Pad, conv1/kernel, conv1/bias, DMT/_0, DMT/_1, DMT/_2)]]
```
The same code (with the same pre-trained model) does work on a linux machine (Intel(R) Xeon(R) CPU E5-2673 v3, running on Microsoft Azure), also with TF built with mkl-dnn.

### Source code / logs

Install MKL 2018 U3 and build MKL-DNN v0.15 from source.

Build TF r1.9 with the following flags:
```
tensorflow_BUILD_ALL_KERNERS=ON
tensorflow_BUILD_CONTRIB_KERNELS=ON
tensorflow_BUILD_SHARED_LIB=ON
tensorflow_ENABLE_GPU=OFF
tensorflow_ENABLE_GRPC_SUPPORT=ON
tensorflow_ENABLE_JEMALLOC_SUPPORT=OFF
tensorflow_ENABLE_MKLDNN_SUPPORT=ON
tensorflow_ENABLE_MKL_SUPPORT=ON
tensorflow_ENABLE_POSITION_INDEPENDENT_CODE=ON
tensorflow_ENABLE_SNAPPY_SUPPORT=ON
tensorflow_OPTIMIZE_FOR_NATIVE_ARCH=ON
tensorflow_WIN_CPU_SIMD_OPTIONS=ON
```

Build passes and generates `tensorflow.dll`. Use keras to save a pre-trained Resnet50 model and attempt inference from C++ (linked with `tensorflow.dll`) as follows:
```
Session* session;
Status status = NewSession(SessionOptions(), &session);

GraphDef graph_def;
status = ReadBinaryProto(Env::Default(), "../graph/resnet50.pb", &graph_def);
status = session->Create(graph_def);

Mat img = imread("../elephant.png"), imgFloat, imgResized;
img.convertTo(imgFloat, CV_32FC3);

Tensor tInput(DT_FLOAT, TensorShape({ 1, 224, 224, 3 }));
auto input_tensor_mapped = tInput.tensor<float, 4>();

Mat imgTensor(224, 224, CV_32FC3);
resize(imgFloat, imgTensor, Size(224, 224));
imgTensor -= Scalar(103.939, 116.779, 123.68);
	
auto source_data = imgTensor.ptr<float>(0);
int width = imgTensor.cols, height = imgTensor.rows, depth=3;
for (int y = 0; y < height; ++y) {
	for (int x = 0; x < width; ++x) {
		for (int c = 0; c < depth; ++c) {
			const float* source_value = source_data + (3*width*y) + (3*x) + (2-c);
			input_tensor_mapped(0, y, x, c) = *source_value;
		}
	}
}

std::vector<std::pair<string, tensorflow::Tensor>> inputs = {
	{ "input_1", tInput },
};

std::vector<tensorflow::Tensor> outputs;
status = session->Run(inputs, { "fc1000/Softmax" }, {}, &outputs);
if (!status.ok()) {
	std::cout << status.ToString() << "\n";
	return 1;
}
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Exact command to reproduce

**Comment by syagev** [comment]
Exact command to reproduce: `Session::run()`

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 91 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by eLvErDe** [comment]
Hello,

This issue is still occuring despite (as usual) TF team closed the bug without a fix:

https://github.com/tensorflow/tensorflow/issues/17494

People are complaining from a wide range of tensorflow versions, up to 1.11.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by sushreebarsa** [comment]
@syagev We see that you are using old version of tensorflow  1.x which is not actively supported, We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions .Please open a new issue in case you face any errors, we will get you the right help . Thanks!
